### ISSUE TITLE
Make maven_jar.artifact available in custom Skylark rules

### ISSUE THREAD

**Issue by simontoens** [bazelbuild/bazel#4876]
### Description of the feature request:
In my Skylark rule, I am looking through all my deps - some of them are maven_jar instances defined in my WORKSPACE file. For those maven_jar instances, I would like to access the value of maven_jar.artifact

### Feature requests: what underlying problem are you trying to solve with this feature?
We are adopting Bazel at our organization, which is currently using Java/Maven/Nexus.  For the foreseeable future, as part of the Bazel build, some jars need to be uploaded to Nexus.  For those jars, we'd like to know their existing Nexus dependencies.  
We are currently hacking around this by parsing the WORKSPACE file ourselves to look at the value of maven_jar.artifact, and we'd like a cleaner solution.

(Also see my question here: https://stackoverflow.com/questions/48470903/access-value-of-maven-jar-artifact-from-skylark-rule)

**Comment by ittaiz** [comment]
Hey,
Two small comments:
1. @Bazel people- We also pursued a similar direction originally and I
think Carmi needed this feature for another reason so maybe these data
points are helpful.
2. @Simon- we invested in this approach and found significant issues in
translating the different semantics (exclusions, provided and maybe more)
from bazel to maven and if you try to also solve the case where a Bazel
artifact depends on a Maven artifact that depends on a “synthesized” jar
from Bazel you’ll find more issues. We decided to go for a parallel track
of maintaining for a whole duplicate copies (with automatic generation of
BUILD files).
HTH
